theme/static/js/hs_file_browser.js

```diff
@@ -125,6 +126,12 @@ function updateSelectionMenuContext() {
         flagDisableDownload = true;
         flagDisableUnzip = true;
         flagDisableGetLink = true;
+        // Multiple folder deletion is not allowed for composite resource
+        // to avoid race condition that can occur with logical file objects especially
+        // if a folder has many files
+        if(resourceType === 'Composite Resource' && selected.length > 1){
```

Comment:
But for file operations inside the UI it was possible to select multiple items. This change would disable deleting when one folder and many files is selected. Wouldn't  it be better to disable delete when at least 2 folders are selected? Also, there is no message indicating that this action is not allowed so it might not be intuitive to users.